pkg/build/metrics/prometheus/metrics.go

```diff
@@ -0,0 +1,124 @@
+		"Counts total teriminal builds by phase",
+		[]string{"phase"},
+		nil,
+	)
+	activeBuildCountDesc = prometheus.NewDesc(
+		buildSubsystem+separator+activeBuildCount,
+		"Show the start time in unix epoch form of running builds by namespace, name, and phase",
+		[]string{"namespace", "name", "phase"},
+		nil,
+	)
+	bc             = buildCollector{}
+	registered     = false
+	failedPhase    = strings.ToLower(string(buildapi.BuildPhaseFailed))
+	errorPhase     = strings.ToLower(string(buildapi.BuildPhaseError))
+	cancelledPhase = strings.ToLower(string(buildapi.BuildPhaseCancelled))
+	completePhase  = strings.ToLower(string(buildapi.BuildPhaseComplete))
+)
+
+type buildCollector struct {
+	lister internalversion.BuildLister
+}
+
+// kube-state-metrics leverages prometheus.Registerer but does not exist in the version of prometheus
+// vendored into origin
+
+func IntializeMetricsCollector(buildLister internalversion.BuildLister) {
+	bc.lister = buildLister
+	// unit tests unearthed duplicate registration with prometheus via multiple calls to new build controller
+	if !registered {
+		prometheus.MustRegister(&bc)
+		registered = true
+	}
+	glog.V(4).Info("build metrics registered with prometheus")
+}
+
+// Describe implements the prometheus.Collector interface.
+func (bc *buildCollector) Describe(ch chan<- *prometheus.Desc) {
+	ch <- terminalBuildCountDesc
+	ch <- activeBuildCountDesc
+}
+
+// Collect implements the prometheus.Collector interface.
+func (bc *buildCollector) Collect(ch chan<- prometheus.Metric) {
+	result, err := bc.lister.List(kselector.Everything())
+
+	if err != nil {
+		glog.V(4).Infof("Collect err %v", err)
+		return
+	}
+
+	// since we do not collect terminal build metrics on a per build basis, collectBuild will return counts
+	// to be added to the total amount posted to prometheus
+	var failed, error, cancelled, complete int
+	for _, b := range result {
+		f, e, cc, cp := bc.collectBuild(ch, b)
+		failed = failed + f
+		error = error + e
+		cancelled = cancelled + cc
+		complete = complete + cp
+	}
+	addCountGauge(ch, terminalBuildCountDesc, failedPhase, float64(failed))
+	addCountGauge(ch, terminalBuildCountDesc, errorPhase, float64(error))
+	addCountGauge(ch, terminalBuildCountDesc, cancelledPhase, float64(cancelled))
+	addCountGauge(ch, terminalBuildCountDesc, completePhase, float64(complete))
+
+	glog.V(4).Infof("reconcileActiveBuildMetrics looked at %d builds", len(result))
+
+}
+
+func addCountGauge(ch chan<- prometheus.Metric, desc *prometheus.Desc, phase string, v float64) {
+	lv := []string{phase}
+	ch <- prometheus.MustNewConstMetric(desc, prometheus.GaugeValue, v, lv...)
+}
+
+func addTimeGauge(ch chan<- prometheus.Metric, b *buildapi.Build, desc *prometheus.Desc) {
+	lv := []string{b.ObjectMeta.Namespace, b.ObjectMeta.Name, strings.ToLower(string(b.Status.Phase))}
+	ch <- prometheus.MustNewConstMetric(desc, prometheus.GaugeValue, float64(b.Status.StartTimestamp.Unix()), lv...)
```

Comment:
ok ... I was checking for nil in `collectBuild` prior to calling `addTimeGauge` ... I knew I was checking for nil somewhere

I did it this way to avoid a function call ... but I can certainly check for nil in both places in case new code is added to call this method